lib/config/fileconf.go

```diff
@@ -2561,12 +2561,14 @@ func (wds *WindowsDesktopService) Check() error {
 		return host.AD
 	})
 
-	if hasAD && wds.LDAP.Addr == "" {
+	ldapConfigOK := wds.LDAP.Addr != "" || (wds.LDAP.Domain != "" && wds.LDAP.LocateServer.Enabled)
+
+	if hasAD && !ldapConfigOK {
 		return trace.BadParameter("if Active Directory hosts are specified in the windows_desktop_service, " +
 			"the ldap configuration must also be specified")
 	}
 
-	if wds.Discovery.BaseDN != "" && wds.LDAP.Addr == "" {
+	if wds.Discovery.BaseDN != "" && !ldapConfigOK {
```

Comment:
I think I missed this in one of my PRs, but maybe you can add the fix while we're here.

`wds.Discovery` is the old way, `wds.DiscoveryConfigs` is the new way.
```suggestion
	if (wds.Discovery.BaseDN != "" || len(wds.DiscoveryConfigs) > 0) && !ldapConfigOK {
```